### DISCUSSION TITLE
How to set up sendgrid as smtp server?

### DISCUSSION THREAD

**Discussion by mastoj** [supabase/supabase#23599] [Questions]
I have followed the instructions on sendgrid here https://docs.sendgrid.com/for-developers/sending-email/integrating-with-the-smtp-api and entered the relevant information in supabase. I have tried both apiKey and apikey as user name and also multiple different ports, but for some reason I can't get send confirmation emails on sign up. Any ideas what might be wrong?

**Comment by epleaner** [comment]
Getting the same thing.

**Comment by romanr** [comment]
Works fine for me. I noticed Sendgrid changed apiKeys recently, maybe you need to regenerate new apiKey.
Here's screenshot of working configuration:
![Arc-Settings  Supabase-0621 0159@2x](https://github.com/supabase/supabase/assets/329079/5fe18dfe-a782-4329-b2b0-2bae2d8018a0)

**Reply by romanr** [reply]
oh another thing I forgot to add, try both addresses; `smtp.sendgrid.net` and `smtp.sendgrid.com`.

**Reply by bangkitdc** [reply]
What about the sender's email and sender name? Does it have to be noreply@ ... ? It doesn't work for me

**Comment by shayne-michael** [comment]
Ran into this same problem, one thing I had to do that wasn't explicit was to setup Sender Authentication in SendGrid. The DNS method is preferred, but if you don't have access to the DNS or it's not working for some reason you can authenticate a single email and be on your way. 
PS I'm using smtp.sendgrid.net at port 587 in Supabase.

<img width="1166" alt="Screenshot 2024-10-31 at 9 25 37 AM" src="https://github.com/user-attachments/assets/34450db8-3eaf-4eeb-a649-3aad5a8add44">
